common/filetoolkitwithundo.cpp

```diff
@@ -226,19 +226,32 @@ ReturnCode FileToolkitWithUndo::createSharedMountPoint(const std::string &path)
         return ReturnCode::FAILURE;
     }
 
-    m_cleanupHandlers.push(new MountCleanUpHandler(path));
+    m_cleanupHandlers.push_back(new MountCleanUpHandler(path));
     log_debug() << "Created shared mount point at " << path;
 
     return ReturnCode::SUCCESS;
 }
 
+bool FileToolkitWithUndo::isPathExist(const std::string path)
+{
+    for (auto element : m_cleanupHandlers) {
+        if (static_cast<FileCleanUpHandler *>(element)->m_path == path) {
+               return true;
+        }
+    }
+    return false;
+}
+
 ReturnCode FileToolkitWithUndo::writeToFile(const std::string &path, const std::string &content)
 {
     auto ret = softwarecontainer::writeToFile(path, content);
     if (isError(ret)) {
         return ret;
     }
-    m_cleanupHandlers.push(new FileCleanUpHandler(path));
+
+    if (!isPathExist(path)) {
```

Comment:
Shouldn't this type of check be done for any cleanup handler? I mean it's good that it is being done here, but shouldn't be done for mounts etc also, to make it more generic?

Perhaps this is a bit over-the-top though, since there is no common way to compare cleanup handlers :/